src/consumer.rs

```diff
@@ -37,6 +37,45 @@ impl<
     }
 }
 
+/// Continuously consumes message from a Queue
+///
+/// A consumer represents the [basic.consume](https://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.consume) AMQP command.
+/// It continuously receives messages from the queue, as opposed to the
+/// [basic.get](https://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.get) command, which
+/// retrieves only a single message.
+///
+/// A consumer is obtained by calling [`Channel::basic_consume`] with the queue name.
+/// New messages from this consumer can be accessed by obtaining the iterator from the consumer.
+/// This iterator returns new messages in the form of a
+/// [`Delivery`] for as long as the consumer is subscribed to the queue.
```

Comment:
Might be wok mentioning basic.cancel to "unsubscribe"

Reply:
While adding this, I stumbled across one thing:

At https://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.consume.consumer-tag it says that a consumer tag is being generated if the field is empty.

I think the API could be improved if the `pub fn tag(&self) -> &ShortString` method is also on the `Consumer` struct itself. I think it's not obvious that the user needs to access the inner struct.
Or is there some reason that it is hidden on the ConsumerInner? Maybe because of the Mutex?

Somehow the rust doc doesn't even generate the documentation for ConsumerInner, so you can only find about the `tag` method when working through the source code.

Reply:
Added Consumre::tag, thanks for noticing it. ConsumerInner shouldn't be public, it was made such as a workarounf when lapin-futures was separated but it's no longer necessary